ResStock 2025 Release 1 Combined Measures

asparke2 opened the pull request:
## Pull Request Description
Adds all of the upgrades to be included in the ResStock 2025 Release 1 dataset.

### Related Pull Requests
[related PRs from different repositories]

### Related Issues
[What issue(s) is the PR addressing]

## Checklist

#### Required:
- [ ] Add to the [changelog_dev.rst file](https://github.com/NREL/resstock/blob/develop/docs/technical_development_guide/source/changelog/changelog_dev.rst)
- [ ] No unexpected regression test changes on CI (comparison artifacts have been checked)
- [ ] Update documentation (one is required)
     - [ ] [Technical reference guide](https://github.com/NREL/resstock/tree/develop/docs/technical_reference_guide)
     - [ ] [Technical development guide](https://github.com/NREL/resstock/tree/develop/docs/technical_development_guide)

#### Optional (not all items may apply):
- [ ] Tests (and test files) have been updated
- [ ] Supporting resource files have been updated (related to resstock-estimation)
  - [ ] [data dictionary](https://github.com/NREL/resstock/tree/develop/resources/data/dictionary)
  - [ ] [source report](https://github.com/NREL/resstock/tree/develop/project_national/resources/source_report.csv)
  - [ ] [options saturation](https://github.com/NREL/resstock/tree/develop/project_national/resources/options_saturations.csv)
  - [ ] [options_lookup](https://github.com/NREL/resstock/blob/develop/resources/options_lookup.tsv)
- [ ] `openstudio tasks.rb update_measures` has been run

Commit:
first commit

Commit:
format

Commit:
SDR options analysis results.

Commit:
SDR integration tests results.

Commit:
Baseline integration tests results.

joseph-robertson committed:
options lookup additions

Commit:
Baseline integration tests results.

joseph-robertson committed:
Merge branch 'develop' into sdr-cchp-dual-fuel2

whiphi92 committed:
duct restriction on typical cold climate

whiphi92 committed:
merging develop into branch

whiphi92 committed:
duct sizing restriction on typical cchp

Commit:
SDR integration tests results.

whiphi92 committed:
testing yml

whiphi92 committed:
Merge branch 'sdr-cchp-dual-fuel2' of https://github.com/NREL/resstock into sdr-cchp-dual-fuel2

whiphi92 committed:
jes buildstock.csv file

katelynstenger committed:
Furnaces yml and modifications

yml, options lookup adjusted, and timeseries analysis

Commit:
SDR options analysis results.

Commit:
SDR integration tests results.

whiphi92 committed:
Merge branch 'develop' into sdr-cchp-dual-fuel2

Commit:
SDR integration tests results.

lixiliu committed:
Merge branch 'develop' into ll/sdr_water_heater

Commit:
SDR integration tests results.

whiphi92 committed:
Merge branch 'develop' into sdr-cchp-dual-fuel2

lixiliu committed:
Merge branch 'develop' into ll/sdr_water_heater

Commit:
updated yml for kestrel

lixiliu committed:
create WH-specific yml

katelynstenger committed:
furnace revisions & options addition

revised logic

katelynstenger committed:
revisions

revs

katelynstenger committed:
Merge branch 'develop' into sdr2025_furnaces

katelynstenger committed:
changes

dev merged

Commit:
update yml

Commit:
Baseline integration tests results.

Commit:
Baseline integration tests results.

lixiliu committed:
Hardcode HPWH options usage bin

Commit:
SDR integration tests results.

Commit:
Baseline integration tests results.

asparke2 committed:
Initial ENERGY STAR V7 Windows upgrade

katelynstenger committed:
revised yml logic

revised upgrades and options lookup

katelynstenger committed:
Merge branch 'sdr2025_furnaces' of https://github.com/NREL/resstock into sdr2025_furnaces

Commit:
Baseline integration tests results.

katelynstenger committed:
revising some yml + options lookup

error in naming and mismatches

Commit:
Baseline integration tests results.

asparke2 committed:
Merge branch 'develop' into energy_star_windows

asparke2 committed:
Merge branch 'cambium_2024' into energy_star_windows

Commit:
first commit

Commit:
SDR options analysis results.

Commit:
SDR integration tests results.

Commit:
Baseline integration tests results.

asparke2 committed:
Adds upgrade.windows column to SDR outputs

lixiliu committed:
update GTWH to include gas access via pool and spa

lixiliu committed:
Update tankless option to UEF

lixiliu committed:
Merge branch 'develop' into ll/sdr_water_heater

lixiliu committed:
readd water heater efficiency

Commit:
SDR options analysis results.

Commit:
SDR integration tests results.

Commit:
Baseline integration tests results.

lixiliu committed:
Add and remove water heater cols from SDR col def

Commit:
merge develop

Commit:
update duct upgrade

Commit:
add heating cooling capacity columns

Commit:
add heating cooling capacity columns

whiphi92 committed:
added upgrades to sdr yml

whiphi92 committed:
Merge branch 'sdr-cchp-dual-fuel2' of https://github.com/NREL/resstock into sdr-cchp-dual-fuel2

whiphi92 committed:
comment error

whiphi92 committed:
Merge branch 'develop' of https://github.com/NREL/resstock into sdr-cchp-dual-fuel2

Commit:
SDR options analysis results.

Commit:
SDR options analysis results.

rajeee committed:
Merge branch 'develop' into sdr_envelope

rajeee committed:
Merge branch 'develop' into ll/sdr_water_heater

rajeee committed:
Merge branch 'develop' into sdr-cchp-dual-fuel2

rajeee committed:
Merge branch 'develop' into sdr2025_furnaces

Commit:
SDR integration tests results.

lixiliu committed:
Add missing saving cols

lixiliu committed:
Merge branch 'll/sdr_water_heater' of github.com:NREL/resstock into ll/sdr_water_heater

Commit:
SDR integration tests results.

asparke2 committed:
Merge branch 'energy_star_windows' into sdr_2025_1_combined

lixiliu committed:
restrict wood/other water heater fuels

asparke2 committed:
Adds upgrades from sdr-cchp-dual-fuel2 manually

asparke2 committed:
Fixes believed typo in anchor naming

asparke2 committed:
Merge remote-tracking branch 'origin/sdr-cchp-dual-fuel2' into sdr_2025_1_combined

asparke2 committed:
Merge remote-tracking branch 'origin/ll/sdr_water_heater' into sdr_2025_1_combined

lixiliu committed:
add check from secondary heating fuel

asparke2 committed:
Merge remote-tracking branch 'origin/ll/sdr_water_heater' into sdr_2025_1_combined

lixiliu committed:
fix gal size

asparke2 committed:
Merge yl/ghp_envelope_0724 YAML changes manually

asparke2 committed:
Merge remote-tracking branch 'origin/sdr_2025_1_combined' into sdr_2025_1_combined

asparke2 committed:
Merge remote-tracking branch 'origin/develop' into sdr_2025_1_combined

asparke2 committed:
Removes unused costs: and lifetime: entries in upgrades

Commit:
upgrade row for air sealing and wall insulation

katelynstenger committed:
updated yml logic based on PR

revised apply logic, renamed package based on changes

asparke2 committed:
Removes unused GHP applicability logic

Applicability matches measure documents.

asparke2 committed:
Merge remote-tracking branch 'origin/sdr_2025_1_combined' into sdr_2025_1_combined

katelynstenger committed:
upgrade furnaces logic

apply logic for references more clear

katelynstenger committed:
rename and refresh shared yml

sdr_upgrade_tm3 yml updated with upgrades as well

katelynstenger committed:
Merge remote-tracking branch 'origin/develop' into sdr2025_furnaces
g Lines starting with '#' will be ignored, and an empty message aborts
merged develop into sdr furnaces branch

lixiliu committed:
update ref AC furnace scenario to replace all

lixiliu committed:
comment out non-applicable options

Commit:
SDR options analysis results.

lixiliu committed:
Add missing option, fix logic

rajeee committed:
Fix typo on sdr column definitions

lixiliu committed:
Add missing cols to sdr def file

lixiliu committed:
typo

asparke2 committed:
Merge remote-tracking branch 'origin/develop' into sdr_2025_1_combined

GitHub-Action committed:
SDR options analysis results.

GitHub-Action committed:
Baseline integration tests results.

GitHub-Action committed:
SDR integration tests results.

lixiliu commented on project_national/sdr_upgrades_tmy3.yml:
```diff
@@ -399,28 +499,159 @@ references:
+      - HVAC Secondary Heating Fuel|Natural Gas
+      - Misc Pool Heater|Natural Gas
+      - Misc Hot Tub Spa|Natural Gas
+      - not: Misc Gas Lighting|None
```
@whiphi92  - Misc Gas <item> should be removed. They are randomly assigned in ResStock and could overestimate the # of homes with gas access.

asparke2 committed:
Remove heating/cooling capacity from savings calculations

asparke2 committed:
Removes misc. gas uses as a valid indicator of natural gas access

Per @lixiliu gas lighting, grill, and fireplace are randomly assigned and could overestimate the number of homes with gas access

asparke2 committed:
Merge remote-tracking branch 'origin/sdr_2025_1_combined' into sdr_2025_1_combined

asparke2 committed:
Merge remote-tracking branch 'origin/develop' into sdr_2025_1_combined

katelynstenger committed:
rename upgrades

renamed the furnace upgrades to be more descriptive.

GitHub-Action committed:
SDR options analysis results.

GitHub-Action committed:
SDR options analysis results.

GitHub-Action committed:
SDR integration tests results.

GitHub-Action committed:
SDR integration tests results.

asparke2 committed:
Merge remote-tracking branch 'origin/sdr2025_furnaces' into sdr_2025_1_combined

asparke2 committed:
Adds timeout to kill long-running models

Avoids whole-job failures (and corresponding re-runs) due to individual long-running models.

GitHub-Action committed:
SDR options analysis results.

GitHub-Action committed:
SDR integration tests results.

asparke2 commented on project_national/sdr_upgrades_tmy3.yml:
```diff
@@ -399,28 +515,385 @@ references:
+
+      - option: HVAC Cooling Partial Space Conditioning|100% Conditioned
+        apply_logic: *typical_cold_climate_logic
+
```
Right now, the upgrade.foo columns for this upgrades:

Typical Cold Climate Ducted Air Source Heat Pump with Detailed Performance Data:
--- upgrade.hvac_heating_efficiency
--- upgrade_name
--- upgrade.hvac_detailed_performance_data
--- upgrade.hvac_cooling_efficiency
--- upgrade.hvac_cooling_partial_space_conditioning

Are you saying that adding this logic doesn't change the applicability/modeling, but does set these fields:

--- upgrade.heating_fuel == Electricity
--- upgrade.hvac_heating_type_and_fuel ==Electricity ASHP

So that if people are looking for post-retrofit units with electric heating, if they filter on upgrade.heating_fuel they'll get the correct subset of units?

asparke2 committed:
Adds GHP borehole count and length to outputs

asparke2 committed:
Merge remote-tracking branch 'origin/sdr_2025_1_combined' into sdr_2025_1_combined

asparke2 committed:
Adds EV driving unmet hours to outputs

asparke2 committed:
Updates GHP borehole length description

whiphi92 reviewed (commented):
Upgrade name change

whiphi92 committed:
Update project_national/sdr_upgrades_tmy3.yml

asparke2 committed:
Set tag-only options for heating fuel and HVAC heating system type

Per @lixiliu these options are informative only and do not change simulation results. The populate the upgrade.heating_fuel and upgrade.hvac_heating_system_type columns in the results for easier filtering.

asparke2 committed:
Updates upgrade names

Removes details specific to modeling and replaces mathematical operators with words and punctuation.

GitHub-Action committed:
SDR options analysis results.

asparke2 committed:
Exclude unmet EV driving hours from savings calculation

aspeake committed:
Add component loads to SDR outputs

aspeake committed:
Merge branch 'sdr_2025_1_combined' of https://github.com/NREL/resstock into sdr_2025_1_combined

asparke2 committed:
Updates SDR upgrade names for fuel oil/propane furnaces and min eff AC and furnaces

asparke2 committed:
Merge remote-tracking branch 'origin/sdr_2025_1_combined' into sdr_2025_1_combined

GitHub-Action committed:
SDR options analysis results.

rajeee committed:
Merge branch 'develop' into sdr_2025_1_combined

rajeee committed:
Merge branch 'develop' into sdr_2025_1_combined

asparke2 committed:
Remove component_load columns from savings calculations

GitHub-Action committed:
SDR integration tests results.
